examples/dmd/generic_csv.cpp

```diff
@@ -155,190 +159,234 @@ int main(int argc, char *argv[])
+            if (myid == 0)
+            {
+                cout << "Loading DMD model #" << window << "." << endl;
+            }
+            dmd[window] = new CAROM::DMD(outputPath + "/window" + to_string(window)); 
         }
+    }
 
-        int overlap_count = 0;
-        for (int idx_snap = 0; idx_snap < num_train_snap[idx_dataset]; ++idx_snap)
+    StopWatch dmd_training_timer, dmd_preprocess_timer, dmd_prediction_timer;
+    double* sample = new double[dim];
+    CAROM::Vector* result = new CAROM::Vector(dim, true);
+
+    if (train)
+    {
+        dmd_training_timer.Start();
+
+        for (int idx_dataset = 0; idx_dataset < npar; ++idx_dataset)
         {
-            string snap = snap_list[idx_snap]; // STATE
-            csv_db->getDoubleArray(string(data_dir) + "/" + par_dir + "/" + snap + "/tval.csv", &tval, 1);
-            string data_filename = string(data_dir) + "/" + par_dir + "/" + snap + "/" + variable + ".csv"; // path to VAR_NAME.csv
-            csv_db->getDoubleArray(data_filename, sample, nelements, idx_state);
-            dmd[curr_window]->takeSample(sample, tval);
-            if (overlap_count > 0)
+            string par_dir = training_par_list[idx_dataset]; // training DATASET
+            if (myid == 0)
             {
-                dmd[curr_window-1]->takeSample(sample, tval);
-                overlap_count -= 1;
+                cout << "Loading samples for " << par_dir << " to train DMD." << endl;
             }
-            if (curr_window+1 < numWindows && idx_snap+1 < num_train_snap[idx_dataset])
+            vector<string> snap_list;
+            csv_db->getStringVector(string(list_dir) + "/" + par_dir + ".csv", snap_list, false);
+
+            int curr_window = 0; 
+            int overlap_count = 0;
+            for (int idx_snap = 0; idx_snap < num_train_snap[idx_dataset]; ++idx_snap)
             {
-                bool new_window = false;
-                if (windowNumSamples < infty)
+                string snap = snap_list[idx_snap]; // STATE
+                double tval = 0.0;
+                csv_db->getDoubleArray(string(data_dir) + "/" + par_dir + "/" + snap + "/tval.csv", &tval, 1);
+                string data_filename = string(data_dir) + "/" + par_dir + "/" + snap + "/" + variable + ".csv"; // path to VAR_NAME.csv
+                csv_db->getDoubleArray(data_filename, sample, nelements, idx_state);
+                dmd[curr_window]->takeSample(sample, tval);
+                if (overlap_count > 0)
                 {
-                    new_window = (idx_snap >= (curr_window+1)*windowNumSamples);
+                    dmd[curr_window-1]->takeSample(sample, tval);
+                    overlap_count -= 1;
                 }
-                else 
-                {
-                    new_window = (tval >= indicator_val[curr_window+1]);
-                }
-                if (new_window)
+                if (curr_window+1 < numWindows && idx_snap+1 < num_train_snap[idx_dataset] && 
```

Comment:
A comment about the condition here, involving the number 10, would be helpful.

Reply:
Add a comment to `parametric_tw_dmd.csv`.